nmosnode/aggregator.py

```diff
@@ -144,60 +371,75 @@ def _process_queue(self):
+                            data = self._node_data["entities"][namespace][res_type][res_key]
                             try:
-                                self.logger.writeInfo("Attempting registration for Node {}"
-                                                      .format(self._registered["node"]["data"]["id"]))
-                                self._SEND("POST", "/{}".format(namespace), data)
-                                self._SEND("POST", "/health/nodes/" + self._registered["node"]["data"]["id"])
-                                self._registered["registered"] = True
-                                if self._mdns_updater is not None:
-                                    self._mdns_updater.P2P_disable()
-
-                            except Exception:
-                                self.logger.writeWarning("Error registering Node: %r" % (traceback.format_exc(),))
-
-                        elif res_key in self._registered["entities"][namespace][res_type]:
-                            data = self._registered["entities"][namespace][res_type][res_key]
-                            try:
-                                self._SEND("POST", "/{}".format(namespace), data)
+                                self._send("POST", self.aggregator, self.aggregator_apiversion,
+                                           "/{}".format(namespace), data)
                             except InvalidRequest as e:
                                 self.logger.writeWarning("Error registering {} {}: {}".format(res_type, res_key, e))
                                 self.logger.writeWarning("Request data: {}".format(data))
-                                del self._registered["entities"][namespace][res_type][res_key]
+                                del self._node_data["entities"][namespace][res_type][res_key]
 
                     elif queue_item["method"] == "DELETE":
                         translated_type = res_type + 's'
                         try:
-                            self._SEND("DELETE", "/{}/{}/{}".format(namespace, translated_type, res_key))
+                            self._send("DELETE", self.aggregator, self.aggregator_apiversion,
+                                       "/{}/{}/{}".format(namespace, translated_type, res_key))
                         except InvalidRequest as e:
                             self.logger.writeWarning("Error deleting resource {} {}: {}"
                                                      .format(translated_type, res_key, e))
                     else:
                         self.logger.writeWarning("Method {} not supported for Registration API interactions"
                                                  .format(queue_item["method"]))
+                except ServerSideError:
+                    self.aggregator = None
+                    self._aggregator_failure = True
+                    self._add_request_to_front_of_queue(queue_item)
                 except Exception:
```

Comment:
This logic doesn't seem to appear in the state diagram. Could it be added as a note?

Reply:
Add to diagram